drivers/sensor/sensor_sim/sensor_sim.c

```diff
@@ -203,21 +255,163 @@ static double generate_pseudo_random(void)
 }
 
 /*
- * @brief Calculates sine from uptime
- * The input to the sin() function is limited to avoid overflow issues
+ * @brief Generate value for acceleration signal toggling between two values on fetch.
+ *
+ * @param chan[in]	Selected sensor channel.
+ * @param out_val[out]	Pointer to the variable that is used to store result.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
+static int generate_toggle(enum sensor_channel chan, double *out_val)
+{
+	static const double amplitude = 20.0;
```

Comment:
Looks like this doesn't need to be static? Or maybe I'm missing something that makes it beneficial to have it as static, I see the same is used in several functions

Reply:
For wave signals, the amplitude is configurable by the application.

In case of toggle, this is just a constant parameter - it's why I preferred to keep it as an static const value with function scope.
This is done to make "toggle" signal on acceleration work in the same way as before. Maybe can we leave it as it is?

Reply:
Thanks, no problem to leave it as is, I was just curious